client/vscode/src/chatActivation.ts

```diff
@@ -434,3 +442,49 @@ function getCurrentEditorParams(): InlineChatParams | undefined {
 
     return params
 }
+async function handleOpenTabRequest(params: any, webviewView: WebviewView) {
+    const mapErrorType = (type: string | undefined): number => {
+        switch (type) {
+            case 'InvalidRequest':
+                return ErrorCodes.InvalidRequest
+            case 'InternalError':
+                return ErrorCodes.InternalError
+            case 'UnknownError':
+            default:
+                return ErrorCodes.UnknownErrorCode
+        }
+    }
+    const webviewMessageId = uuidv4()
+    const webViewResponsePromise = new Promise<UiMessageResultParams | undefined>((resolve, reject) => {
+        const timeout = setTimeout(() => {
+            reject(new Error('Request timed out'))
+        }, 30000)
+
+        const messageHandler = (message: any) => {
+            if (typeof webviewMessageId !== 'string' && message.webviewMessageId === webviewMessageId) {
```

Comment:
How is it  working? Don't we need to pass `webviewMessageId` back from `chat.ts`?